File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        visited = set()
        while n != 1 and n not in visited:
            visited.add(n)
            res = 0
            for digit in str(n):
                digit = int(digit)
                res += digit * digit
            n = res
            res = 0
        return n == 1
        

File: 202-happy-number/happy-number.ts
function isHappy(n: number): boolean {
    const visited = new Set<number>()
    while (n !== 1 && !(visited.has(n)) ) {
        visited.add(n)
        let sum = 0
        while (n > 0) {
            let digit = n%10
            sum += digit*digit
            n = Math.floor(n/10)
        }
        n = sum
        sum = 0
    }

    return n===1
};